tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,1157 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test authentication REST API """
+
+import os
+import subprocess
+import http.client
+import json
+import time
+import pytest
+from pathlib import Path
+
+__author__ = "Yash Tatkondawar"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+PROJECT_ROOT = ""
+SCRIPTS_DIR_ROOT = ""
+TEMPLATE_NAME = "template.json"
+SENSOR_VALUE = 12.25
+# TODO : pass package_build_version to setup script from conftest.py
+package_build_version = "nightly"
+HTTP_SOUTH_SVC_NAME = "SOUTH_HTTP"
+HTTP_SOUTH_SVC_NAME_1 = "SOUTH_HTTP_1"
+ASSET_NAME = "auth"
+PASSWORD_TOKEN = None
+CERT_TOKEN = None
+
+
+@pytest.fixture
+def set_project_root():
+    global PROJECT_ROOT, SCRIPTS_DIR_ROOT
+    # This  gives the path of directory where FogLAMP is cloned. test_file < packages < python < system < tests < ROOT
+    PROJECT_ROOT = Path(__file__).parent.parent.parent.parent.parent
+    SCRIPTS_DIR_ROOT = "{}/tests/system/python/packages/data/".format(PROJECT_ROOT)
+
+
+def call_fogbench(wait_time):
+    execute_fogbench = 'cd {}/extras/python ;python3 -m fogbench -t $FOGLAMP_ROOT/data/tests/{} ' \
+                       '-p http -O 10'.format(PROJECT_ROOT, TEMPLATE_NAME)
+    exit_code = os.system(execute_fogbench)
+    assert 0 == exit_code
+    time.sleep(wait_time)
+
+
+def add_south_http(foglamp_url, name, TOKEN):
+    payload = {"name": name, "type": "south", "plugin": "http_south", "enabled": True}
+    post_url = "/foglamp/service"
+    conn = http.client.HTTPConnection(foglamp_url)
+    conn.request("POST", post_url, json.dumps(payload), headers={"authorization": TOKEN})
+    res = conn.getresponse()
+    assert 200 == res.status, "ERROR! POST {} request failed".format(post_url)
+    res = res.read().decode()
+    r = json.loads(res)
+    return r
+
+
+def generate_json(asset_name):
+    subprocess.run(["cd $FOGLAMP_ROOT/data && mkdir -p tests"], shell=True, check=True)
+
+    fogbench_template_path = os.path.join(
+        os.path.expandvars('${FOGLAMP_ROOT}'), 'data/tests/{}'.format(TEMPLATE_NAME))
+    with open(fogbench_template_path, "w") as f:
+        f.write(
+            '[{"name": "%s", "sensor_values": '
+            '[{"name": "sensor", "type": "number", "min": %f, "max": %f, "precision": 2}]}]' % (
+                asset_name, SENSOR_VALUE, SENSOR_VALUE))
+
+
+@pytest.fixture
+def change_to_auth_mandatory_any(foglamp_url, wait_time):
+    # Wait for foglamp server to start
+    time.sleep(wait_time)
```

Comment:
for all such methods, I believe this statement (with comment) should be after `'/foglamp/restart'` call